package.json

```diff
@@ -3,6 +3,10 @@
   "version": "2.0.3",
   "description": "Uber-fast unique id generation, for Node.js and the browser",
   "main": "hyperid",
+  "files": [
+    "hyperid.js",
+    "index.d.ts"
+  ],
```

Comment:
I’m not in favor of this. Tests are an integral part of this module.

Reply:
It does not make sense to me, please let me explain.

For me - part of how I evaluate a package is I check what's the size and if it ships files that do not belong to production.

Also -

When a module is installed in node_modules, it is installed without it's dev-dependencies.
This means that you cannot really run the tests, although you can read them.

If you'd expect for docs - what I'd do is provide links from within the README.md which by-definision, ships always.

In order to run the tests you have to cd node_modules/hyperid && npm i - which is something you're not encouraged to do. If you do it - you'll get node_modules/hyperid/node_modules which is different from what you get from npm i on your project root - in terms of duplicates and extranous dependencies. You're interfering with something that should have been encapsulated - you basically should let npm manage it. - so It's an anti-pattern.

mm.
Can I please sincerely ask, do you really feel that people should run tests of a dependency from inside node_modules?

If you're not convinced - we can add "tests", to "files": [, but can we agree that all there rest do not belong in the package?
For sure - the .travis-ci.yaml...

Reply:
> Can I please sincerely ask, do you really feel that people should run tests of a dependency from inside node_modules?

Tests are a form of documentation. By looking at them, a developer can understand how a project is used. As a result, they are part of a release as much as docs is.

I'm pretty happy to have test in there and drop the rest.

Reply:
personally, I don't believe that docs belong in the package sent to production, but you da boss :grin: